File: back-end/models/Game.go
package models

import (
	"time"

	"github.com/RafaelMoreira96/game-beating-project/utils"
	"github.com/go-playground/validator/v10"
)

type Game struct {
	IdGame      uint       `gorm:"primaryKey" json:"id_game"`
	NameGame    string     `json:"name_game"`
	Developer   string     `json:"developer"`
	GenreID     uint       `json:"genre_id"`
	Genre       Genre      `gorm:"foreignKey:GenreID" json:"genre"`
	ConsoleID   uint       `json:"console_id"`
	Console     Console    `gorm:"foreignKey:ConsoleID" json:"console"`
	DateBeating utils.Date `json:"date_beating"`
	TimeBeating float64    `json:"time_beating"`
	ReleaseYear string     `json:"release_year"`
	PlayerID    uint       `json:"player_id"`
	Player      Player     `gorm:"foreignKey:PlayerID" json:"-"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
}

func (game *Game) Validate() error {
	validate := validator.New()
	return validate.Struct(game)
}

File: front-end/web/src/app/models/game.ts
import { Console } from "./console";
import { Genre } from "./genre";

export class Game {
  id_game: number;
  name_game: string;
  developer: string;
  genre_id: number;
  genre?: Genre;  // Agora é opcional
  console_id: number;
  console?: Console;  // Agora é opcional
  date_beating: string;
  time_beating: number;
  release_year: string;
  player_id: number;
  created_at: Date;
  updated_at: Date;

  constructor() {
    this.id_game = 0;
    this.name_game = '';
    this.developer = '';
    this.genre_id = 0;
    this.genre = new Genre;  // Inicializa como undefined
    this.console_id = 0;
    this.console = new Console;  // Inicializa como undefined
    this.date_beating = '';
    this.time_beating = 0;
    this.release_year = '';
    this.player_id = 0;
    this.created_at = new Date();
    this.updated_at = new Date();
  }
}
